CMakeLists.txt

```diff
@@ -265,10 +265,14 @@ endif (PGINDENT)
 
 option(USE_OPENSSL "Enable use of OpenSSL if available" ON)
 
-# Check if PostgreSQL has OpenSSL enabled.
+# Check if PostgreSQL has OpenSSL enabled by inspecting pg_config --configure.
 # Right now, a Postgres header will redefine an OpenSSL function if Postgres is not installed --with-openssl,
 # so in order for TimescaleDB to compile correctly with OpenSSL, Postgres must also have OpenSSL enabled.
-file(STRINGS ${PG_INCLUDEDIR}/pg_config.h PG_USE_OPENSSL REGEX "#define USE_OPENSSL [01]")
+execute_process(
+  COMMAND ${PG_CONFIG} --configure
+  OUTPUT_VARIABLE PG_CONFIGURE_FLAGS
+  OUTPUT_STRIP_TRAILING_WHITESPACE)
+string(REGEX MATCH "--with-openssl" PG_USE_OPENSSL ${PG_CONFIGURE_FLAGS})
```

Comment:
You should wrap `${PG_CONFIGURE_FLAGS}` in quotes so if by some chance it's empty, this check still works. Otherwise, tested on a version with no OpenSSL and works as expected :+1:

Reply:
Fixed